src/subdomains/supporting/payment/services/transaction-helper.ts

```diff
@@ -489,12 +489,14 @@ export class TransactionHelper implements OnModuleInit {
     if (isBuy) {
       if (
         user?.status === UserStatus.NA &&
-        ((user?.wallet.amlRule === AmlRule.RULE_2 && user?.userData.kycLevel < KycLevel.LEVEL_30) ||
-          (user?.wallet.amlRule === AmlRule.RULE_3 && user?.userData.kycLevel < KycLevel.LEVEL_50) ||
-          (user?.wallet.amlRule === AmlRule.RULE_6 &&
+        (([user?.wallet.amlRule, from.amlRule, to.amlRule].includes(AmlRule.RULE_2) &&
+          user?.userData.kycLevel < KycLevel.LEVEL_30) ||
+          ([user?.wallet.amlRule, from.amlRule, to.amlRule].includes(AmlRule.RULE_3) &&
+            user?.userData.kycLevel < KycLevel.LEVEL_50) ||
+          ([user?.wallet.amlRule, from.amlRule, to.amlRule].includes(AmlRule.RULE_6) &&
             paymentMethodIn === FiatPaymentMethod.CARD &&
             user?.userData.kycLevel < KycLevel.LEVEL_30) ||
-          (user?.wallet.amlRule === AmlRule.RULE_7 &&
+          ([user?.wallet.amlRule, from.amlRule, to.amlRule].includes(AmlRule.RULE_7) &&
             paymentMethodIn === FiatPaymentMethod.CARD &&
             user?.userData.kycLevel < KycLevel.LEVEL_50))
       )
```

Comment:
Maybe we can move this to AML helper? Would keep the logic (which is the same) together.